main/mafia.hs

```diff
@@ -502,6 +509,17 @@ mafiaScript :: File -> [Argument] -> EitherT MafiaError IO ()
 mafiaScript file args =
   firstT MafiaScriptError $ runScript file args
 
+mafiaExec :: [Argument] -> EitherT MafiaError IO ()
+mafiaExec args =
+  liftCabal $ cabal_ "exec" $ fixupArgs args
+  where
+    fixupArgs = \case
+      [] -> [] -- Should not happen.
+      [x] -> [x] -- A command without arguments.
+      -- Insert `--` between the command and its arguments so cabal doesn't
+      -- mess with them.
+      (x:xs) -> x : "--" : xs
```

Comment:
I'm good with this, just a minor nit, I'm trying to get rid of uses of `where` from this codebase, so if you don't mind can we change it to something like:
```hs
mafiaExec :: [Argument] -> EitherT MafiaError IO ()
mafiaExec args =
  let
    fixupArgs = \case
      [] ->
        [] -- Should not happen.
      [x] ->
        [x] -- A command without arguments.
      -- Insert `--` between the command and its arguments so cabal doesn't
      -- mess with them.
      x : xs ->
        x : "--" : xs
  in
    liftCabal $ cabal_ "exec" $ fixupArgs args
```